### ISSUE TITLE
Release 1.2.1

### ISSUE THREAD

**Issue by Thrameos** [jpype-project/jpype#918]
- [x] Start from the release branch
      ``git checkout release``
- [x] Make a new branch for the release cycle
      ``git checkout -b releases/{version}``
- [x] Merge the current master with the release
      ``git pull origin master``
- [x] Start a release
      ``bumpversion release``
- [x] Edit doc/CHANGELOG.rst
- [x] Send the release to be evaluated
      ``git push``
- [x] Verify CI on azure
- [x] Manually trigger a ``jpype.release`` on azure
      If successful, download the artifacts for publication.
- [x] Advance the release pointer
      ``git checkout release``
      ``git merge releases/<version>``
- [x] Publish the release
  - Add draft release on github
  - Attach the artifacts to the release.
- [x] Start master on a new cycle
  - Use a PR to pull release back to master
  - ``git checkout master``
  - ``git checkout -b cycle``
  - ``git merge release``
  - ``bumpversion patch``
  - Use PR to insert the cycle in master

**Comment by Thrameos** [comment]
@marscher I am guessing this should be a patch release.   Most of the stuff on it was bugs and missing files from 1.2.0.   Nothing changed in the API.

**Comment by marscher** [comment]
Thanks for pushing this forward. It is indeed a micro release. Just a question regarding release branches, whats the purpose of having a branch for every past release? Isn't a git tag sufficient for that? Was the intention to backport changes to older releases (for which we do not have enough man power in my opinion)?

**Comment by Thrameos** [comment]
Yes, the branches are there just in case we end up in a case where some feature is very troublesome and we have to support a previous release with a back port.  We can remove the older ones one we have a complete release. (Such as dropping 1.1.x)  

There were a few times that I was adding a major feature where we also needed to bug fix a previous release.  So I believe this is a good practice.  Not that we will use it often, but it is very nice to have when it is needed.

**Comment by marscher** [comment]
It is good practice to back port bugfixes for important releases. I just cannot recall a version for which we did it in the past. Eventually I just didn't noticed this.

On January 2, 2021 11:47:07 PM UTC, Karl Nelson <notifications@github.com> wrote:
>Yes, the branches are there just in case we end up in a case where some
>feature is very troublesome and we have to support a previous release
>with a back port.  We can remove the older ones one we have a complete
>release. (Such as dropping 1.1.x)  
>
>There were a few times that I was adding a major feature where we also
>needed to bug fix a previous release.  So I believe this is a good
>practice.  Not that we will use it often, but it is very nice to have
>when it is needed.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/jpype-project/jpype/issues/918#issuecomment-753544856

-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by Thrameos** [comment]
Records say we did backporting for a few critical bugs from 0.7.4 to 0.7.5.   I anticipate we will end up in a similar situation from version 1.2.x to 1.3.x or 2.0.x (depending on how we want to do the epypj release).  

I am about 90% finished on the reverse bridge.   But there serious issue will be how to execute the testing.   When we are testing the reverse bridge everything (including the testing) has to be originated from within Java to be valid.   So my guess is we will have to run a testng bench and then somehow merge the test results from both Python and Java.  I am sure I will find a lot of bugs once the testing framework is done, but right now all I have been able to do is create a Python context, import a few packages and call some methods.

**Comment by Thrameos** [comment]
Windows had a bug with url specifications.   I believe it is fixed now so that I can complete the release, but it is too late tonight to finish.

**Comment by marscher** [comment]
Wouldn't it be easier to create these branches of the release tags, once we make the decision to backport a change? This way we would need fewer branches, which we need to keep track of. I'm glad you're making such good progress with the reverse bridge, this will be a killer feature!

**Comment by marscher** [comment]
Release is published on PyPI and conda-forge.

**Comment by Thrameos** [comment]
Thanks.  Hopefully this won't have problems
